src/Numbers.jsx

```diff
@@ -0,0 +1,49 @@
+import { useState } from "react"
+function Numbers({state, ans})
+{
+    const [currentState, newState] = state
+    const [isBeforeSeparator, toggleBeforeSeparator] = useState(true)
+    const changeState = (number) => {
+        if (currentState === 0) {
+            newState(number)
+            return
+        }
+        newState(`${currentState}${number}`)
+    }
+
+    const clear = () =>{
+        toggleBeforeSeparator(true)
+        newState(0)
+    }
+
+    const buttonArray = []
+    for (let i = 1; i< 10; i++){
+        buttonArray.push(<button onClick={() => changeState(i)}>{i}</button>)
```

Comment:
not complete - it's missing 0, although I see you are hardcoding 0 in the return statement

better to define a const array numbers = ["1", "2", etc...] and then use nubmers.map() directly in your return JSX block